### PR TITLE
[MacCoss] Issue 34193: Allow search for small molecules in UI

### PR DESCRIPTION
**Pull request by labkey-bpatel**
#### Rationale
[MacCoss] Issue [34193](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=34193): Allow search for small molecules in UI

#### Changes
* Add 'Small Molecule Search' tab to the existing Panorama Web Search.
* Add underlying search queries and create search results view for display.
* Handle rendering of the search results.
* Update URL, handle Refresh/Back button.
* Remove ext4 dependency on Protein and Peptide views.
* Update column ordering on Protein and Peptide grids.

### CODE REVIEW

File: panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp [hunk 1]
```diff
@@ -256,17 +297,45 @@
             case "protein":
                 clearInputFromProteinTab();
                 break;
+            case "smallMolecule":
+                clearInputFromProteinTab();
```
Comment (labkey-jeckels) [inline panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp]:
```suggestion
                clearInputFromSmallMoleculeTab();
```

File: panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp [hunk 2]
```diff
@@ -228,6 +264,11 @@
         document.getElementById(exactPeptideMatchesItemId).checked = false;
     };
 
+    let clearInputFromSmallMoleculeTab = function () {
+        document.getElementById(smallMoleculeItemId).value = ""
```
Comment (labkey-jeckels) [inline panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp]:
```suggestion
        document.getElementById(smallMoleculeItemId).value = "";
```

File: panoramapublic/resources/views/smallMoleculeSearchResults.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        let wp = new LABKEY.QueryWebPart({
+            renderTo: webpart.wrapperDivId,
+            frame: 'portal',
+            title: 'Search criteria : ' + smallMolecule + ' & exact match: ' + exactMatch ,
+            schemaName: 'targetedms',
+            queryName: 'smallMoleculeSearchResults',
+            showFilterDescription: false,
+            parameters: {smallMolecule : smallMolecule, exactMatch: exactMatch},
+            containerFilter: LABKEY.Query.containerFilter.currentAndSubfolders,
+            success: function () {
```
Comment (labkey-jeckels) [inline panoramapublic/resources/views/smallMoleculeSearchResults.html]: I don't think this `success` handler is needed.

File: panoramapublic/resources/views/proteinSearchResults.html [hunk 4]
```diff
@@ -24,12 +17,7 @@
             parameters: {proteinLabel : label, exactMatch: exactMatch},
             containerFilter: LABKEY.Query.containerFilter.currentAndSubfolders,
             success: function () {
```
Comment (labkey-jeckels) [inline panoramapublic/resources/views/proteinSearchResults.html]: Shouldn't need this now-empty function

File: panoramapublic/resources/views/peptideSearchResults.html [hunk 5]
```diff
@@ -24,12 +17,7 @@
             parameters: {peptideSequence : peptideSequence, exactMatch: exactMatch},
             containerFilter: LABKEY.Query.containerFilter.currentAndSubfolders,
             success: function () {
-                if (Ext4.get('search-indicator')) {
```
Comment (labkey-jeckels) [inline panoramapublic/resources/views/peptideSearchResults.html:27]: Shouldn't need this now-empty function

**Review by labkey-jeckels — APPROVED** [review]

File: panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp (hunk 1 above)
Comment (labkey-bpatel) [reply panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp]: done

File: panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp (hunk 2 above)
Comment (labkey-bpatel) [reply panoramapublic/src/org/labkey/panoramapublic/view/search/panoramaWebSearch.jsp]: done

File: panoramapublic/resources/views/proteinSearchResults.html (hunk 4 above)
Comment (labkey-bpatel) [reply panoramapublic/resources/views/proteinSearchResults.html]: good point, now removed

**Comment by labkey-bpatel** [comment]
All tests passed! yay!

**Pull request merged**